assets/javascripts/discourse/initializers/chat-setup.js

```diff
@@ -124,6 +127,156 @@ export default {
+              return this.channel.title;
+            }
+
+            get prefixIcon() {
+              return "hashtag";
+            }
+
+            get prefixIconColor() {
+              return this.channel.chatable.color;
+            }
+
+            get prefixIconBadge() {
+              return this.channel.chatable.read_restricted && "lock";
+            }
+
+            get suffixIcon() {
+              return this.channel.unread_count > 0 && "circle";
+            }
+
+            get suffixCssClass() {
+              return this.channel.unread_mentions > 0 ? "urgent" : "unread";
+            }
+          };
+
+          const SidebarChatSection = class extends BaseCustomSidebarSection {
+            @tracked sectionLinks = A([]);
+
+            constructor({ sidebar }) {
+              super(...arguments);
+
+              this.sidebar = sidebar;
+              this.chatService = container.lookup("service:chat");
+              this.sidebar.appEvents.on(
+                "chat:refresh-channels",
+                this._refreshChannels
+              );
+              this.sidebar.appEvents.on(
+                "chat:navigated-to-full-page",
+                this._refreshChannels
+              );
+              this.currentUser = getOwner(this).lookup("current-user:main");
+            }
+
+            willDestroy() {
+              this.sidebar.appEvents.off(
+                "chat:refresh-channels",
+                this._refreshChannels
+              );
+              this.sidebar.appEvents.off(
+                "chat:navigated-to-full-page",
+                this._refreshChannels
+              );
+            }
+
+            @bind
+            _refreshChannels() {
+              const newSectionLinks = [];
+              this.chatService.getChannels().then((channels) => {
+                channels.publicChannels.forEach((channel) => {
+                  const trackingState =
+                    this.currentUser.chat_channel_tracking_state[channel.id];
+                  if (channel === this.chatService.activeChannel) {
+                    channel.unread_count = 0;
+                    channel.unread_mentions = 0;
+                  } else {
+                    channel.unread_count = trackingState.unread_count;
+                    channel.unread_mentions = trackingState.unread_mentions;
+                  }
+                  newSectionLinks.push(new SidebarChatSectionLink({ channel }));
+                });
+              });
+              this.sectionLinks = newSectionLinks;
+            }
+
+            get name() {
+              return I18n.t("chat.chat_channels");
+            }
+
+            get title() {
+              return I18n.t("chat.chat_channels");
+            }
+
+            get text() {
+              return I18n.t("chat.chat_channels");
+            }
+
+            get actions() {
+              const currentUser = getOwner(this).lookup("current-user:main");
+              const actions = [
```

Comment:
It feels slightly odd in the API that everything is a class and then suddenly it becomes a class POJO